crates/wasmtime/src/trap.rs

```diff
@@ -264,6 +264,11 @@ impl Trap {
         }
     }
 
+    /// Displays the error reason for this trap.
+    pub fn display_reason(&self) -> &impl fmt::Display {
```

Comment:
Could this return `impl Display` so if necessary we can change this in the future to returning something that isn't necessarily stored on `Self`?